pkg/config/environment_detection.go

```diff
@@ -99,11 +99,18 @@ func detectFeatures() {
 	if IsAutoconfigEnabled() {
 		detectContainerFeatures(newFeatures)
 		excludedFeatures := Datadog.GetStringSlice("autoconfig_exclude_features")
-		for _, ef := range excludedFeatures {
-			delete(newFeatures, Feature(ef))
+		for _, f := range excludedFeatures {
+			delete(newFeatures, Feature(f))
+		}
+
+		includedFeatures := Datadog.GetStringSlice("autoconfig_include_features")
+		for _, f := range includedFeatures {
+			newFeatures[Feature(f)] = struct{}{}
 		}
 
 		log.Infof("Features detected from environment: %v", newFeatures)
+	} else {
+		log.Warnf("Deactivating Autoconfig will disable most components. It's recommended to use autoconfig_exclude_features and autoconfig_include_features to activate/deactivate features selectively")
```

Comment:
Checks can always be configured independently by adding/removing the configuration files (a new option will come to drop built-in files), so it would for cases like this.

Activating/de-activating some components within the feature is indeed not possible, it would require introducing parameters for each sub-component (tagger, hostname, etc.) which are not existing and probably not very useful.

Reply:
thanks for clarifying this, the new option to drop built-in files will improve the situation indeed!